qiskit_experiments/framework/composite/composite_analysis.py

```diff
@@ -206,16 +211,42 @@ def _marginalized_component_data(self, composite_data: List[Dict]) -> List[List[
                         sub_data["counts"] = datum["counts"]
                 if "memory" in datum:
                     if composite_clbits is not None:
-                        sub_data["memory"] = (
-                            np.array(datum["memory"])[composite_clbits[i]]
-                        ).tolist()
+                        # level 2
+                        if f_memory is not None:
+                            idx = slice(composite_clbits[i][0], composite_clbits[i][-1] + 1)
+                            sub_data["memory"] = [shot[::-1][idx][::-1] for shot in f_memory]
```

Comment:
I forgot about the reversed bit indexing in my slice example. I think you can remove the need to reverse and re-reverse with `::-1` if you  build the slice with that already in mind. I tried some quick tests and I *think* this is correct:
```python
idx = slice(-1 - composite_qubits[i][-1], -composite_qubits[i][0] or None)
sub_data["memory"] = [shot[idx] for shot in f_memory]
```
The `None` case is needed to handle when a qubit[0] = 0

Reply:
That's better thanks. See here: [f9eefda](https://github.com/Qiskit/qiskit-experiments/pull/801/commits/f9eefda2ebfab9469ccd56e01c676971d60049b3)